app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingReservationPresenter.java

```diff
@@ -1,17 +1,85 @@
 package com.example.parkingsystemjava.mvp.presenter;
 
+import com.example.parkingsystemjava.entity.Reservation;
+import com.example.parkingsystemjava.listener.ListenerDateTime;
 import com.example.parkingsystemjava.mvp.contract.ParkingReservationContract;
+import com.example.parkingsystemjava.utils.Constants;
+import com.example.parkingsystemjava.utils.ReservationErrorCodes;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
 
-public class ParkingReservationPresenter implements ParkingReservationContract.ReservationActivityPresenter {
+public class ParkingReservationPresenter implements ParkingReservationContract.ParkingReservationPresenter {
 
-    private ParkingReservationContract.ReservationActivityView view;
+    private final ParkingReservationContract.ParkingReservationView view;
+    private final ParkingReservationContract.ParkingReservationModel model;
 
-    public ParkingReservationPresenter(ParkingReservationContract.ReservationActivityView view) {
+    public ParkingReservationPresenter(ParkingReservationContract.ParkingReservationModel model, ParkingReservationContract.ParkingReservationView view) {
+        this.model = model;
         this.view = view;
     }
 
     @Override
-    public void showDatePicker() {
-        view.showEntryDatePicker();
+    public void showDatePicker(ListenerDateTime listenerDateTime, boolean dateSelector) {
+        view.showDatePicker(listenerDateTime, dateSelector);
+    }
+
+    @Override
+    public void addReservation(Reservation reservation, int parkingLot) {
+        model.addReservation(reservation, parkingLot);
+    }
+
+    @Override
+    public void setEntryDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showEntryDateSelected(stringDate);
+    }
+
+    @Override
+    public void setExitDate(Calendar date) {
+        String stringDate = getStringFromDate(date);
+        view.showExitDateSelected(stringDate);
+    }
+
+    @Override
+    public boolean validateReservation(Reservation reservation, int parkingLot) {
+        ReservationErrorCodes reservationCode = model.validateReservation(reservation, parkingLot);
+        if (reservationCode.equals(ReservationErrorCodes.OK)) {
+            view.showConfirmationMessage(reservationCode);
+            return true;
+        } else {
+            view.showConfirmationMessage(reservationCode);
+            return false;
+        }
+    }
+
+    @Override
+    public void validateAndSaveReservation(String entryDate, String exitDate, String keyCode, String parkingLot) {
+        Reservation reservation = new Reservation();
```

Comment:
please call only one method to set all the reservation necessary info

Reply:
done, implemented new constructor in Reservation